src/components/NePaginator.vue

```diff
@@ -0,0 +1,167 @@
+<!--
+  Copyright (C) 2023 Nethesis S.r.l.
+  SPDX-License-Identifier: GPL-3.0-or-later
+-->
+
+<script setup lang="ts">
+import { computed } from 'vue'
+import { range } from 'lodash-es'
+
+export type NePaginatorProps = {
+  currentPage: number
+  totalPages: number
+  previousLabel: string
+  nextLabel: string
+}
+
+const props = defineProps<NePaginatorProps>()
```

Comment:
When using Typescript to declare props, the linter sometimes find it hard to do type checking, e.g. detecting required props not passed to the component. I suggest to use the  "old standard" way to define props, specifying `type` and `required` or `default` for every prop

Reply:
Sure, though I've still left the `NePaginatorProps` type and export for all outer components which want to use the type of the paginator's props!